### PR TITLE
Add shared implementation of dispatching logic

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix https://github.com/aspnet/EntityFramework/issues/5182

cc @NTaylorMullen @Eilon

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Tools.Cli/project.json [hunk 1]
```diff
@@ -17,6 +17,7 @@
     "Microsoft.DotNet.Cli.Utils": "1.0.0-*",
     "Microsoft.DotNet.ProjectModel": "1.0.0-*",
     "Microsoft.Extensions.CommandLineUtils": "1.0.0-*",
+    "Microsoft.Extensions.DotnetToolDispatcher.Sources": "1.0.0-*",
```
Comment (NTaylorMullen) [inline src/Microsoft.EntityFrameworkCore.Tools.Cli/project.json]: build time dependency

File: src/Microsoft.EntityFrameworkCore.Tools/project.json [hunk 2]
```diff
@@ -23,6 +23,7 @@
       ],
       "dependencies": {
         "Microsoft.EntityFrameworkCore.Tools.Cli": "1.0.0-*",
+        "Microsoft.Extensions.DotnetToolDispatcher.Sources": "1.0.0-*",
```
Comment (NTaylorMullen) [inline src/Microsoft.EntityFrameworkCore.Tools/project.json]: ditto

**Comment by NTaylorMullen** [comment]
I dont see use of the `IsDispatcher` method. How do you determine that?

:watch:

**Comment by natemcmaster** [comment]
"IsDispatcher" is implied by the separate assemblies that "`Ms.EfCore.Tools`" contains dispatcher and "`Ms.EfCore.Tools.Cli`" contains the inside-man.

File: src/Microsoft.EntityFrameworkCore.Tools/project.json (hunk 2 above)
Comment (natemcmaster) [inline src/Microsoft.EntityFrameworkCore.Tools/project.json]: Ah, will fix

**Comment by natemcmaster** [comment]
Updated

**Comment by NTaylorMullen** [comment]
:shipit:

**Pull request merged**